File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map <int, int> mp; 
  vector <int> return_vec; 
  int carry;
  for (int i = 0; i < nums.size(); i++){
    carry = target - nums[i];
    if (mp.count(nums[i])){
        return_vec.push_back(mp[nums[i]]);
      return_vec.push_back(i);
      return return_vec;
    }
    else{
      mp[carry] = i;
    }
  }
  return return_vec;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # bs #
        '''
        -1 0 2 4 6 8 
        '''
        lp = 0 
        rp = len(nums) - 1 
        found = False 
        while lp <= rp and not found:
            mp = (lp + rp) // 2 
            if target < nums[mp]:
                rp = mp - 1 
            elif target > nums[mp]:
                lp = mp + 1 
            else:
                found = True
        if found:
            return (lp + rp) // 2  
        return lp 